### PR TITLE
Add css var for header webkit backdrop filter

### PR DESCRIPTION
**Pull request by bgoncal**
## Proposed change

This adds a CSS variable to customize -webkit-backdrop-filter for header.

https://github.com/home-assistant/frontend/assets/5808343/9b2c5e43-f243-44da-b872-7590b45f9788

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/layouts/home-assistant-main.ts [hunk 1]
```diff
@@ -177,6 +177,7 @@ export class HomeAssistantMain extends LitElement {
         -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
         --mdc-drawer-width: 56px;
         --mdc-top-app-bar-width: calc(100% - var(--mdc-drawer-width));
+        --backdrop-filter: unset;
```
Comment (bramkragten) [inline src/layouts/home-assistant-main.ts:180]: No need to set this

File: src/panels/lovelace/hui-root.ts [hunk 2]
```diff
@@ -908,6 +908,7 @@ class HUIRoot extends LitElement {
           position: fixed;
           top: 0;
           width: var(--mdc-top-app-bar-width, 100%);
+          -webkit-backdrop-filter: var(--backdrop-filter, unset);
```
Comment (bramkragten) [inline src/panels/lovelace/hui-root.ts:911]: Can we name the variable `header-backdrop-filter`

**Review by bramkragten — COMMENTED** [review]
You only implemented this for the Lovelace dashboard now, we should probably do it more general?

**Comment by bgoncal** [comment]
> You only implemented this for the Lovelace dashboard now, we should probably do it more general?

Do I need to replicate the change in many places or there is a way to declare this variable once and apply to all headers?

**Comment by bramkragten** [comment]
Most use `ha-top-app-bar-fixed`, so implementing it there will get most cases.

File: src/panels/lovelace/hui-root.ts [hunk 3]
```diff
@@ -908,6 +908,7 @@ class HUIRoot extends LitElement {
           position: fixed;
           top: 0;
           width: var(--mdc-top-app-bar-width, 100%);
+          -webkit-backdrop-filter: var(--header-backdrop-filter, unset);
```
Comment (bramkragten) [inline src/panels/lovelace/hui-root.ts:911]: Also add the non vendor prefixed key:
```suggestion
          -webkit-backdrop-filter: var(--header-backdrop-filter, unset);
          backdrop-filter: var(--header-backdrop-filter, unset);
```

File: src/components/ha-top-app-bar-fixed.ts [hunk 4]
```diff
@@ -27,6 +27,9 @@ export class HaTopAppBarFixed extends TopAppBarFixedBase {
         padding-inline-start: 20px;
         padding-inline-end: initial;
       }
+      .header {
```
Comment (bramkragten) [inline src/components/ha-top-app-bar-fixed.ts:30]: I don't think there is an element that has the `header` class? Have you tested this?

Comment (Nezz) [reply src/components/ha-top-app-bar-fixed.ts:30]: I think this is where this should go:
https://github.com/home-assistant/frontend/blob/35cf3063cb09e583c607239cd3658621a2e04312/src/panels/lovelace/hui-root.ts#L904

File: src/layouts/hass-error-screen.ts [hunk 5]
```diff
@@ -68,6 +68,7 @@ class HassErrorScreen extends LitElement {
           padding: 8px 12px;
           pointer-events: none;
           background-color: var(--app-header-background-color);
+          -webkit-backdrop-filter: var(--header-backdrop-filter, unset);
```
Comment (Nezz) [inline src/layouts/hass-error-screen.ts:71]: The `app` prefix is used for the background. Should it be added to this variable too?

Comment (bgoncal) [reply src/layouts/hass-error-screen.ts:71]: My idea is to make this available for all devices, not app specific

File: src/components/ha-top-app-bar-fixed.ts (hunk 4 above)
Comment (bgoncal) [reply src/components/ha-top-app-bar-fixed.ts:30]: > I don't think there is an element that has the `header` class? Have you tested this?

Oh sorry, I got confused here, if I want to add it directly to ha-top-app-bar-fixed can I just remove ".header"?

File: src/layouts/hass-error-screen.ts (hunk 5 above)
Comment (Nezz) [reply src/layouts/hass-error-screen.ts:71]: I think app here refers to "web app" and not "mobile app".

Comment (bgoncal) [reply src/layouts/hass-error-screen.ts]: Oh, sounds good then, I've updated

File: src/components/ha-top-app-bar-fixed.ts (hunk 4 above)
Comment (bgoncal) [reply src/components/ha-top-app-bar-fixed.ts:30]: Screenshot 2024-04-09 at 15 13 14
I think I am adding it twice

Comment (bgoncal) [reply src/components/ha-top-app-bar-fixed.ts:30]: @Nezz but would it then go to all headers in HA?

Comment (Nezz) [reply src/components/ha-top-app-bar-fixed.ts:30]: Oh right, that way it'd only get added to the dashboards. Add it under this one:
https://github.com/home-assistant/frontend/blob/38f8b4e0877da7ca8e19e2650424ea016f096480/src/components/ha-top-app-bar-fixed.ts#L18
and this one:
https://github.com/home-assistant/frontend/blob/38f8b4e0877da7ca8e19e2650424ea016f096480/src/components/ha-top-app-bar.ts#L18

File: src/panels/lovelace/hui-root.ts [hunk 6]
```diff
@@ -908,6 +908,8 @@ class HUIRoot extends LitElement {
           position: fixed;
           top: 0;
           width: var(--mdc-top-app-bar-width, 100%);
+          -webkit-backdrop-filter: var(--app-header-backdrop-filter, unset);
+          backdrop-filter: var(--app-header-backdrop-filter, unset);
```
[the remark this answers is not in the document]
Comment (Nezz) [reply src/panels/lovelace/hui-root.ts:912]: Also note that both properties should be added everywhere.

[the remark this answers is not in the document]
Comment (bgoncal) [reply src/panels/lovelace/hui-root.ts:912]: Indeed, I first need to know where exactly they need to be and where they don't

[the remark this answers is not in the document]
Comment (bgoncal) [reply src/panels/lovelace/hui-root.ts]: I removed from other places and kept it in these 3 files you see in the changes, what do you think?

**Comment by Nezz** [comment]
>You only implemented this for the Lovelace dashboard now, we should probably do it more general?

After having a look around, it seems that only these two panels support transparency in their headers:
https://github.com/search?q=repo%3Ahome-assistant%2Ffrontend+calc%28var%28--header-height%29+%2B+env%28safe-area-inset-top%29%29%3B&type=code

The rest of the views (logbook, settings, notifications, etc) don't allow the content to slide under the header. Therefore adding the `backdrop-filter` there would be just slow down the rendering.

**Comment by Nezz** [comment]
Nice, I haven't found the developer tools

**Review by bramkragten — APPROVED** [review]

**Comment by nayzm** [comment]
Thank you for working on this. 

Just wondering would it be possible to implement some sort of support for [kiosk-mode](https://github.com/maykar/kiosk-mode)? On mobile devices with `app-header` hidden, the background doesn't stretch underneath the status icons on iOS, so it just shows a solid colour.

**Comment by bgoncal** [comment]
> Thank you for working on this. 
> 
> 
> 
> Just wondering would it be possible to implement some sort of support for [kiosk-mode](https://github.com/maykar/kiosk-mode)? On mobile devices with `app-header` hidden, the background doesn't stretch underneath the status icons on iOS, so it just shows a solid colour.

You can workaround this by setting a app-header-background-color right? This would change the color of the status bar too

**Comment by nayzm** [comment]
> > Thank you for working on this.
> > Just wondering would it be possible to implement some sort of support for [kiosk-mode](https://github.com/maykar/kiosk-mode)? On mobile devices with `app-header` hidden, the background doesn't stretch underneath the status icons on iOS, so it just shows a solid colour.
> 
> You can workaround this by setting a app-header-background-color right? This would change the color of the status bar too

It doesn't unfortunatly, even when setting a rgba for `app-header-background-color`.

4F012F27-A142-4AAF-8D80-0C3E056299DE_4_5005_c

**Comment by bgoncal** [comment]
Keep an eye on this PR, it may help dealing with this: https://github.com/home-assistant/iOS/pull/2757

**Pull request merged**